### PR TITLE
Apply tdd

### PR DESCRIPTION
**Pull request by mohamedgamaleldin999999**
In this PR, I am trying to apply TDD following these requirements:

Create a class called Solver.
Create a method called [factorial](https://en.wikipedia.org/wiki/Factorial) that takes one argument, an integer N, and returns the factorial for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
When N is divisible by 3, return "fizz".
When N is divisible by 5, return "buzz".
When N is divisible by 3 and 5, return "fizzbuzz".
Any other case, return N as a string (e.g. say N is 7 then return "7").
Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    before(:each) do
+      @solver = Solver.new
+    end
+
+    it 'should calculate factorial of ceratin input as integer number ' do
+      expect(@solver.factorial(2)).to eq 2
+      expect(@solver.factorial(5)).to eq 120
+      expect(@solver.factorial(10)).to eq 3_628_800
+      expect(@solver.factorial(15)).to eq 1_307_674_368_000
+    end
```
Comment (AbayomiOlaoye) [inline spec/solver_spec.rb:14]: - [ ] You have implemented brilliant processes here but you seem to be missing the need to test the code against unintended operations on negative numbers. Kindly write a test for such a negative case to raise exception error as contained in the requirement :point_down: 

image

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @Team :wave: :wave: , 

Good job with this milestone so far! What a brilliant attempt!!
There are just a few more issues to work on before moving on to the next project but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of Gitflow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Great teamwork :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 

## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
